src/app/coursegrab/utils/scraper.py

```diff
@@ -0,0 +1,71 @@
+import requests
+import bs4
+
+ROOT_URL = "https://classes.cornell.edu"
+
+
+def current_semester():
+    req = requests.get(ROOT_URL)
+    req.raise_for_status()
+    return req.url.strip("/").split("/")[-1]
+
+
+def all_subject_codes():
+    req = requests.get(ROOT_URL)
+    roster_bs4 = bs4.BeautifulSoup(req.text, "html.parser")
+    subject_tags = roster_bs4.select(".browse-subjectcode")
+    return [str(tag.getText()) for tag in subject_tags]
+
+
+def get_course_status(subject_code, catalog_num):
+    semester = current_semester()
+    subject_url = "http://classes.cornell.edu/browse/roster/" + semester + "/subject/" + subject_code
+    subject_page = requests.get(subject_url)
+    subject_page.raise_for_status()
+    subject_bs4 = bs4.BeautifulSoup(subject_page.text, "html.parser")
+    catalog_tags = subject_bs4.find_all("strong", class_="tooltip-iws")
+    for tag in catalog_tags:
+        catalog_code = int(tag.getText().strip())
+        if catalog_code == catalog_num:
+            section = tag.parent.parent.parent
+            status = section.find_all("li", class_="open-status")[0].contents[0].contents[0]["class"][-1]
+            if "open-status-open" in status:
+                return "open"
```

Comment:
can we make an [enum](https://docs.python.org/3/library/enum.html) for this

Reply:
what benefit does using enums give us rather than just a constants file?